n2y/plugins/internallinks.py

```diff
@@ -0,0 +1,68 @@
+import typing
+import uuid
+import urllib.parse
+
+from n2y.blocks import Block, ChildPageBlock
+from n2y.logger import logger
+from n2y.page import Page
+from n2y.rich_text import TextRichText
+from n2y.utils import header_id_from_text
+
+
+def get_notion_id_from_href(href: str) -> typing.Optional[str]:
```

Comment:
`... -> typing.Optional[str]:` can be simplified to `...-> str | None` if you want. It's a preference choice, but just making sure you're aware.

Reply:
That syntax was introduced in 3.10 and the setup.py file indicates n2y might be intended to be compatible with earlier Python versions.

Reply:
Gotcha. That needs to get updated, actually. n2y is supported for up to 3.11 not just 3.9

Reply:
@sHermanGriffiths note that n2y is used with the website so if we use features that require a new version of python we may force the website to update their python too if that makes sense.

Reply:
Good point!